.github/workflows/maven.yml

```diff
@@ -1,771 +1,562 @@
-      - name: Upload the build
-        uses: actions/upload-artifact@v3
-        with:
-          name: MacOS-x86_64-Build-JDK8
-          path: /Users/runner/work/Brotli4j/
-
-  MacOS-x86_64-Build-JDK11:
-    runs-on: macos-latest
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up JDK 11
-        uses: actions/setup-java@v3
-        with:
-          distribution: 'corretto'
-          java-version: 11
-          architecture: x64
-      - name: Build with Maven
-        run: mvn -B --show-version -ntp --file pom.xml clean package
+      - name: Checkout
+        uses: actions/checkout@v4
 
-  MacOS-x86_64-Build-JDK17:
-    runs-on: macos-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: Set up JDK 17
-        uses: actions/setup-java@v3
-        with:
-          distribution: 'corretto'
-          java-version: 17
-          architecture: x64
-      - name: Build with Maven
-        run: mvn -B --show-version -ntp --file pom.xml clean package
+      - name: Pre Setup JDK
+        run: |
+          sudo rm -rf /opt/hostedtoolcache/Java_*/*
 
-  MacOS-x86_64-Build-JDK21:
-    runs-on: macos-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: Set up JDK 21
-        uses: actions/setup-java@v3
+      - name: Setup JDK
+        uses: actions/setup-java@v4
         with:
           distribution: 'corretto'
-          java-version: 21
-          architecture: x64
-      - name: Build with Maven
-        run: mvn -B --show-version -ntp --file pom.xml clean package
-
-  Linux-x86_64-Build-JDK8:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: build centos6
-        run: docker-compose -f docker/docker-compose.yml run build
-      - name: Upload the build
-        uses: actions/upload-artifact@v3
-        with:
-          name: Linux-x86_64-Build-JDK8
-          path: /home/runner/work/Brotli4j/
-
-  Linux-x86_64-Build-JDK11:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: build centos6
-        run: docker-compose -f docker/docker-compose11.yml run build
-
-  Linux-x86_64-Build-JDK17:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: build centos6
-        run: docker-compose -f docker/docker-compose17.yml run build
+          java-version: ${{ matrix.java_version }}
+          architecture: 'x64'
+          check-latest: true
+
+      - name: Setup Ubuntu Build
+        if: ${{ matrix.architecture.ubuntu_version }}
+        run: |
+          docker build \
+              --platform linux/amd64 \
+              --build-arg "DOCKER_IMAGE=amd64/ubuntu:${{ matrix.architecture.ubuntu_version }}" \
+              --build-arg "TARGET_GCC_PACKAGE=${{ matrix.architecture.gcc_package_name }}" \
+              --build-arg "TARGET_GCC_EXEC=${{ matrix.architecture.gcc_exec_name }}" \
+              --tag linux-buildimage:${{ matrix.architecture.name }} \
+              docker/build/
```

Comment:
The actions file is now giant. Can we dump these into a file?

Reply:
It's not meant to be small, it's meant be easy to follow.

Putting the docker commands into extra files will just lead to a bunch of scripts with random parameters that are hard to follow.